Amplify/Categories/Predictions/PredictionsCategory+ClientBehavior.swift

```diff
@@ -9,81 +9,38 @@ import Foundation
-                       options: options,
-                       listener: listener)
-    }
-
-    /// Translate the text to the language specified.
-    /// - Parameter textToTranslate: The text to translate
-    /// - Parameter language: The language of the text given
-    /// - Parameter targetLanguage: The language to which the text should be translated
-    /// - Parameter options: Parameters to specific plugin behavior
-    /// - Parameter listener: Triggered when the event occurs
-    @discardableResult
-    public func convert(textToTranslate: String,
-                        language: LanguageType?,
-                        targetLanguage: LanguageType?,
-                        options: PredictionsTranslateTextRequest.Options? = nil,
-                        listener: PredictionsTranslateTextOperation.ResultListener?
-    ) -> PredictionsTranslateTextOperation {
-        plugin.convert(textToTranslate: textToTranslate,
-                           language: language,
-                           targetLanguage: targetLanguage,
-                           options: options,
-                           listener: listener)
-    }
-
-    /// Transcribe audio to text
-    /// - Parameter speechToText: The url of the audio to be transcribed
-    /// - Parameter listener: Triggered when the event occurs
-    /// - Parameter options: Parameters to specific plugin behavior
-    @discardableResult
-    public func convert(speechToText: URL,
-                        options: PredictionsSpeechToTextRequest.Options?,
-                        listener: PredictionsSpeechToTextOperation.ResultListener?
-    ) -> PredictionsSpeechToTextOperation {
-        plugin.convert(speechToText: speechToText, options: options, listener: listener)
-    }
-
     /// Detect contents of an image based on `IdentifyAction`
     /// - Parameter type: The type of image detection you want to perform
     /// - Parameter image: The image you are sending
     /// - Parameter options: Parameters to specific plugin behavior
     /// - Parameter listener: Triggered when the event occurs
-    @discardableResult
-    public func identify(type: IdentifyAction,
-                         image: URL,
-                         options: PredictionsIdentifyRequest.Options? = nil,
-                         listener: PredictionsIdentifyOperation.ResultListener?
-    ) -> PredictionsIdentifyOperation {
-        plugin.identify(type: type,
-                        image: image,
-                        options: options,
-                        listener: listener)
+    public func identify<Output>(
+        _ request: Predictions.Identify.Request<Output>,
+        in image: URL,
+        options: Predictions.Identify.Options? = nil
+    ) async throws -> Output {
+        try await plugin.identify(request, in: image, options: options)
+    }
+
+    public func convert<Input, Options, Output>(
+        _ request: Predictions.Convert.Request<Input, Options, Output>,
```

Comment:
🤔  so the `request` has `Options` but the method also have `Options`?

Reply:
Yup, the `options` arg in the method is the generic `Options` because it depends on the type of convert `request`.